web3_proxy/src/frontend/users/payment_stripe.rs

```diff
@@ -0,0 +1,240 @@
+use crate::app::Web3ProxyApp;
+use crate::errors::{Web3ProxyError, Web3ProxyErrorContext, Web3ProxyResponse, Web3ProxyResult};
+use crate::frontend::authorization::{
+    login_is_authorized, Authorization as Web3ProxyAuthorization,
+};
+use crate::frontend::users::authentication::register_new_user;
+use anyhow::Context;
+use axum::{
+    extract::Path,
+    headers::{authorization::Bearer, Authorization},
+    response::IntoResponse,
+    Extension, Json, TypedHeader,
+};
+use axum_client_ip::InsecureClientIp;
+use axum_macros::debug_handler;
+use entities::{
+    balance, increase_on_chain_balance_receipt, rpc_key, stripe_increase_balance_receipt, user,
+};
+use ethers::abi::AbiEncode;
+use ethers::types::{Address, Block, TransactionReceipt, TxHash, H256};
+use hashbrown::{HashMap, HashSet};
+use http::StatusCode;
+use log::{debug, error, info, trace};
+use migration::sea_orm::prelude::Decimal;
+use migration::sea_orm::{
+    self, ActiveModelTrait, ActiveValue, ColumnTrait, EntityTrait, IntoActiveModel, ModelTrait,
+    QueryFilter, QuerySelect, TransactionTrait,
+};
+use migration::{Expr, OnConflict};
+use payment_contracts::ierc20::IERC20;
+use payment_contracts::payment_factory::{self, PaymentFactory};
+use serde_json::json;
+use std::num::NonZeroU64;
+use std::sync::Arc;
+use stripe::Webhook;
+
+/// `GET /user/stripe/deposits` -- Use a bearer token to get the user's balance and spend.
+///
+/// - shows a list of all deposits, including their chain-id, amount and tx-hash
+#[debug_handler]
+pub async fn user_stripe_deposits_get(
+    Extension(app): Extension<Arc<Web3ProxyApp>>,
+    TypedHeader(Authorization(bearer)): TypedHeader<Authorization<Bearer>>,
+) -> Web3ProxyResponse {
+    let (user, _semaphore) = app.bearer_is_authorized(bearer).await?;
+
+    let db_replica = app.db_replica().context("Getting database connection")?;
+
+    // Filter by user ...
+    let receipts = stripe_increase_balance_receipt::Entity::find()
+        .filter(increase_on_chain_balance_receipt::Column::DepositToUserId.eq(user.id))
+        .all(db_replica.as_ref())
+        .await?;
+
+    // Return the response, all except the user ...
+    let response = json!({
+        "user": Address::from_slice(&user.address),
+        "deposits": receipts,
+    });
+
+    Ok(Json(response).into_response())
+}
+
+/// `POST /user/balance/stripe/` -- Process a confirmed txid to update a user's balance.
```

Comment:
Comment should be about stripe, not txids

Reply:
changed